timeslots/timeslots.go

```diff
@@ -1,13 +1,53 @@
 package timeslots
 
+import (
+	"sort"
+	"time"
+)
+
+type TimeslotBase struct {
+	OwnerId    string `json:"ownerId"`
+	BookedById string `json:"bookedById"`
+	Booked     bool   `json:"booked"`
+}
+
 type Timeslot struct {
-	OwnerId    string     `json:"ownerId"`
-	BookedById string     `json:"bookedById"`
-	Time       TimePeriod `json:"time"`
-	Booked     bool       `json:"booked"`
+	TimeslotBase
+	Time []TimePeriod `json:"time"`
+}
+
+type TimeslotInDb struct {
+	TimeslotBase
+	TimeFrom int64 `json:"timeFrom"`
+	TimeTo   int64 `json:"timeTo"`
 }
 
 type TimePeriod struct {
-	From string `json:"from"`
-	To   string `json:"to"`
+	From time.Time `json:"from"`
+	To   time.Time `json:"to"`
+}
+
+func isTimePeriodValid(timeperiod TimePeriod) bool {
+	return timeperiod.From.Before(timeperiod.To)
+}
+
+func sortByTimeFrom(timeperiods []TimePeriod) {
+	sort.Slice(timeperiods, func(i, j int) bool {
+		return timeperiods[i].From.Before(timeperiods[j].From)
+	})
+}
+
+func areTimePeriodsOverlapping(timeperiods []TimePeriod) bool {
```

Comment:
todo: not sure where in the code to use this function yet.

Reply:
```
sort.Slice(timeperiods, func(i, j int) bool {
		return timeperiods[i].From.Before(timeperiods[j].From)
	})
	
        prev := tt[0]
	for i:= 1; i < len(tt)); i++ {
	     cur := tt[i]
	     if !prev.To.Before(cur.From) {
	            return false
	      }
	      prev = cur
        }
        return true
```